### PR TITLE
fix: consider all 'devel' builds as 'dev' builds

### PR DESCRIPTION
**Pull request by Emyrk**
If CI needs to be distinguished from a dev build, we should add a different pre-release tag for those builds.

### CODE REVIEW

File: buildinfo/buildinfo.go [hunk 1]
```diff
@@ -65,16 +73,21 @@ func Version() string {
 func VersionsMatch(v1, v2 string) bool {
 	// Developer versions are disregarded...hopefully they know what they are
 	// doing.
-	if strings.HasPrefix(v1, develPrefix) || strings.HasPrefix(v2, develPrefix) {
+	if IsDevVersion(v1) || IsDevVersion(v2) {
```
Comment (mafredri) [inline buildinfo/buildinfo.go:76]: I think this logic made sense for `v0.0.0`, but for non-`v0.0.0` we know approximately the version in question and we could require the minor version match. Or is there a use-case for returning a match for e.g. `v1.1.1-devel` and `v1.2.3` as tested below?

Comment (Emyrk) [reply buildinfo/buildinfo.go:76]: I was thinking along the same lines. I will change it so only v0.0.0 matches all, and other versions do check major minor as expected.

File: buildinfo/buildinfo.go [hunk 2]
```diff
@@ -71,9 +71,9 @@ func Version() string {
 // disregarded. If it detects that either version is a developer build it
 // returns true.
 func VersionsMatch(v1, v2 string) bool {
-	// Developer versions are disregarded...hopefully they know what they are
-	// doing.
-	if IsDevVersion(v1) || IsDevVersion(v2) {
+	// If no version is attached, then it is a dev build outside of CI. The version
+	// will be disregarded... hopefully they know what they are doing.
+	if strings.Contains(v1, noVersion) || strings.Contains(v2, noVersion) {
 		return true
```
Comment (Emyrk) [inline buildinfo/buildinfo.go:77]: This should match the old behavior now @mafredri

**Review by mafredri — APPROVED** [review]

File: buildinfo/buildinfo.go (hunk 1 above)
Comment (Emyrk) [reply buildinfo/buildinfo.go]: This should be pretty much the previous behavior.

**Pull request merged**